protocol/hello/hello.go

```diff
@@ -90,11 +90,11 @@ func (h *Handler) handleNewStream(s net.Stream) {
 
 	switch err := h.processHelloMessage(from, &hello); err {
 	case ErrBadGenesis:
-		log.Warningf("genesis cid: %s does not match: %s, disconnecting from peer: %s", &hello.GenesisHash, h.genesis, from)
+		log.Debugf("genesis cid: %s does not match: %s, disconnecting from peer: %s", &hello.GenesisHash, h.genesis, from)
```

Comment:
Should the "bad hello message" case above also disconnect?

Reply:
I don't know. I believe an error for that case indicates a failure to decode the cbor message, my gut says disconnecting is the correct behaviour. I can add that here as well as the metics since it will be a small change.

Reply:
Yeah, I'd say it makes sense to disconnect if you send a bad hello. Not sending a hello at all should be okay though, but if youre gonna play the game, better follow the rules.